src/helpers/merging.py

```diff
@@ -0,0 +1,51 @@
+from constants import OUTPUT_COLUMNS
+
+NUMERIC_COLUMNS = ['Pop Tested',
+                   'Pop Tested Positive',
+                   'Pop Tested Negative',
+                   'Pop Deaths',
+                   'Staff Tested',
+                   'Staff Tested Positive',
+                   'Staff Tested Negative',
+                   'Staff Deaths']
+TEXT_COLUMNS = ['Source',
+                'Notes']
+
+assert set(NUMERIC_COLUMNS).issubset(set(OUTPUT_COLUMNS))
+assert set(TEXT_COLUMNS).issubset(set(OUTPUT_COLUMNS))
+
+
+def get_value(row, column):
+  try:
+    return int(row[column])
+  except (KeyError, ValueError):
+    return ''
+
+
+def merge_column(column, rows_with_sources, sources_used):
+  # Use the largest count since it was probably recorded latest in the day
+  rows_with_values = list(filter(lambda row_with_source: '' != get_value(row_with_source[0], column),
+                                 rows_with_sources))
+  row = None
+  if rows_with_values:
+    row, source = max(rows_with_values,
+                      key=lambda row_with_source: get_value(row_with_source[0], column))
+
+  if row:
+    sources_used.add(source)
+
+  return get_value(row, column) if row else ''
+
+
+def merge_rows(rows_with_sources):
+  merged_row = {}
+  sources_used = set()
+
+  for column in NUMERIC_COLUMNS:
+    merged_row[column] = merge_column(column, rows_with_sources, sources_used)
+
+  for column in TEXT_COLUMNS:
+    merged_row[column] = ','.join([row[column] for row, source_id in rows_with_sources
+                                   if row.get(column) and source_id in sources_used])
```

Comment:
So this conditional (`if source_id in sources_used`) depends on `sources_used` having been populated by the previous for loop, right? I think that's somewhat non-intuitive. Why would it be that the presence of a source in the numeric columns would dictate its presence in the text columns?

Reply:
At least for the existing text columns, `Source` and `Notes`, it doesn't seem to make sense to include that information from an input row in an output row if the numeric values from that input row weren't ultimately used in the output row. If that doesn't make sense to you, let's file an issue and circle up with C.

Reply:
Okay I buy that. Let's proceed as is for now!